edx_rest_framework_extensions/auth/jwt/cookies.py

```diff
@@ -23,11 +36,47 @@ def get_decoded_jwt(request):
     """
     Grab jwt from jwt cookie in request if possible.
 
+    Note: May return the decoded authorization header JWT in some cases.
+
     Returns a decoded jwt dict if it can be found.
     Returns None if the jwt is not found.
     """
-    jwt_cookie = request.COOKIES.get(jwt_cookie_name(), None)
+    # use cached jwt if it exists
+    decoded_jwt_from_auth = get_decoded_jwt_from_auth(request)
+    if decoded_jwt_from_auth:
+        return decoded_jwt_from_auth
 
+    # check if jwt cookie exists
+    jwt_cookie = request.COOKIES.get(jwt_cookie_name(), None)
     if not jwt_cookie:
         return None
-    return configured_jwt_decode_handler(jwt_cookie)
+
+    # Custom attribute can be used to determine priority of caching.
+    # Note: Not all IDAs may support increment_custom_attribute
+    set_custom_attribute('jwt_cookie_processed', True)
+    increment_custom_attribute('jwt_cookie_processed_count')
+
+    # .. toggle_name: DISABLE_NEW_JWT_COOKIE_PROCESSING
+    # .. toggle_implementation: DjangoSetting
+    # .. toggle_default: False
+    # .. toggle_description: Rollout toggle to disable the new Jwt cookie processing code, in case of a bug.
+    # .. toggle_use_cases: temporary
+    # .. toggle_creation_date: 2022-06-14
+    # .. toggle_target_removal_date: 2022-07-01
+    # .. toggle_tickets: ARCHBOM-2129
+    if getattr(settings, 'DISABLE_NEW_JWT_COOKIE_PROCESSING', False):
+        return configured_jwt_decode_handler(jwt_cookie)
+
+    try:
+        drf_request = request
+        if not isinstance(request, DrfRequest):
+            drf_request = DrfRequest(request, parsers=api_settings.DEFAULT_PARSER_CLASSES)
+        user_and_auth = JwtAuthentication().authenticate(drf_request)
+        if user_and_auth is not None:
+            return configured_jwt_decode_handler(user_and_auth[1])
+
+        logger.info('get_decoded_jwt: Failed to decode Jwt cookie.')
+    except Exception as ex:  # pylint: disable=broad-except
+        logger.info(f'get_decoded_jwt: Unknown error decoding Jwt cookie. {ex!r}')
```

Comment:
@feanil: In digging up PRs from the time that mobile switched to JWTs, I found this one.
1. I'm wondering if these "info" logs should really be debug logs.
2. I'm wondering it this could use your new error handling code. Or, even simpler, if this were switched to debug we could provide the full exception.

Reply:
I think using the new error extraction code here would be useful and then I would leave them as `INFO` since the debug logs don't get published by default in production.

Reply:
@feanil: Oh geez. I didn't realize I decided to close this PR instead of merging it. And, I don't remember what I was trying to do to begin with. It possibly had something to do with the JWT mobile work, but I decided at the time this wasn't needed, so let's just trust earlier me. :)